# In Class Exercise: Python

In [25]:
import numpy as np
x = np.linspace(1,10,num=5)
print(x)

[  1.     3.25   5.5    7.75  10.  ]


## Print the Array and the exp(x)

Output should be something like:

Hello World: exp( [1. 3.25 5.5 ]) = [ result ]

In [29]:
print("Hello World: exp({}) = {}").format(x, np.exp(x))

Hello World: exp([  1.     3.25   5.5    7.75  10.  ]) = [  2.71828183e+00   2.57903399e+01   2.44691932e+02   2.32157241e+03
   2.20264658e+04]


## Map a function to x

Map the function $\exp(a)*\exp(-a)$ to `x` 

In [27]:
result = map(lambda a: np.exp(a)*np.exp(-a),x)

## Filter the result from the previous Map

Only display the results where $\left|1.0 - r \right| < 10^{-20}$

Print the number of elements in the result

In [28]:
def func(x):
    if abs(1.0 - x) < 10**-20:
        return x

filter(func, result)

[1.0, 1.0, 1.0]